From: Sebastian Keller <skeller@gnome.org>
Date: Sun, 19 Jun 2022 16:24:59 +0200
Subject: Adwaita: Hide SSD box-shadow border on maximized windows

The 1px shadow was showing up on adjacent monitors when a SSD window was
maximized. Additionally this was causing mutter to skip direct scanout
of these windows, because they extend beyond the screen.

Related: https://gitlab.gnome.org/GNOME/mutter/-/issues/2304
Origin: upstream, 3.24.35, commit:4873f63c07e62bd727746aa2d52cf1c13af6c9fd
---
 gtk/theme/Adwaita/_common.scss | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gtk/theme/Adwaita/_common.scss b/gtk/theme/Adwaita/_common.scss
index 198690e..dc02b49 100644
--- a/gtk/theme/Adwaita/_common.scss
+++ b/gtk/theme/Adwaita/_common.scss
@@ -4601,6 +4601,8 @@ decoration {
   // just doing borders, wm draws actual shadows
   .ssd & { box-shadow: 0 0 0 1px $_wm_border; }
   .ssd &:backdrop { box-shadow: 0 0 0 1px $_wm_border_backdrop; }
+  .ssd.maximized &,
+  .ssd.maximized &:backdrop { box-shadow: none; }
 
   .csd.popup & {
     border-radius: $menu_radius;
